src/components/views/rooms/MessageComposerInput.js

```diff
@@ -839,35 +834,26 @@ export default class MessageComposerInput extends React.Component {
             contentText = contentText.substring(4);
             // bit of a hack, but the alternative would be quite complicated
             if (contentHTML) contentHTML = contentHTML.replace(/\/me ?/, '');
-            sendHtmlFn = this.client.sendHtmlEmote;
-            sendTextFn = this.client.sendEmoteMessage;
+            sendHtmlFn = ContentHelpers.makeHtmlEmote;
+            sendTextFn = ContentHelpers.makeEmoteMessage;
         }
 
-        if (quotingEv) {
-            const cli = MatrixClientPeg.get();
-            const room = cli.getRoom(quotingEv.getRoomId());
-            const sender = room.currentState.getMember(quotingEv.getSender());
-
-            const {body/*, formatted_body*/} = quotingEv.getContent();
-
-            const perma = makeEventPermalink(quotingEv.getRoomId(), quotingEv.getId());
-            contentText = `${sender.name}:\n> ${body}\n\n${contentText}`;
-            contentHTML = `<a href="${perma}">Quote<br></a>${contentHTML}`;
-
-            // we have finished quoting, clear the quotingEvent
-            dis.dispatch({
-                action: 'quote_event',
-                event: null,
-            });
-        }
+        const quotingEv = RoomViewStore.getQuotingEvent();
+        const content = quotingEv ? Reply.getRelationship(quotingEv) : {};
+        // we have finished quoting, clear the quotingEvent
+        // TODO maybe delay this until the event actually sends?
```

Comment:
Why not use the existing dispatch `message_sent`?

Reply:
`message_sent` seems like it has the intent to fire for file uploads (which it did not previously) so does not make sense for this whilst it is limited to textual messages due to fallback requirement.